pkg/operation/cloudbotanist/awsbotanist/kube2iam.go

```diff
@@ -0,0 +1,177 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsbotanist
+
+import (
+	"fmt"
+	"strings"
+
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	awsclient "github.com/gardener/gardener/pkg/client/aws"
+	"github.com/gardener/gardener/pkg/operation"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/operation/shoot"
+	"github.com/gardener/gardener/pkg/operation/terraformer"
+
+	awsv1alpha1 "github.com/gardener/gardener-extensions/controllers/provider-aws/pkg/apis/aws/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// DISCLAIMER:
+// The whole file is deprecated. We are keeping kube2iam for a short amount of time for backwards compatibility reasons,
+// however, end-users are asked to deploy kube2iam on their own if they want to use it. Also, we haven't updated the version
+// since long and won't do that anymore.
+// Everything related to kube2iam will be removed in the future.
+
+const terraformerPurposeKube2IAM = "kube2iam"
+
+// DeployKube2IAMResources creates the respective IAM roles which have been specified in the Shoot manifest
+// addon section. Moreover, some default IAM roles will be created.
+// +deprecated
+func DeployKube2IAMResources(o *operation.Operation) error {
+	if !o.Shoot.Kube2IAMEnabled() {
+		return DestroyKube2IAMResources(o)
+	}
+
+	values, err := generateTerraformKube2IAMConfig(o.Shoot)
+	if err != nil {
+		return err
+	}
+
+	tf, err := o.NewShootTerraformer(terraformerPurposeKube2IAM)
+	if err != nil {
+		return err
+	}
+
+	return tf.
+		SetVariablesEnvironment(generateTerraformKube2IAMVariablesEnvironment(o.Shoot.Secret)).
+		InitializeWith(o.ChartInitializer("aws-kube2iam", values)).
+		Apply()
+}
+
+// DestroyKube2IAMResources destroy the kube2iam resources created by Terraform. This comprises IAM roles and
+// policies.
+// +deprecated
+func DestroyKube2IAMResources(o *operation.Operation) error {
+	tf, err := o.NewShootTerraformer(terraformerPurposeKube2IAM)
+	if err != nil {
+		return err
+	}
+	return tf.
+		SetVariablesEnvironment(generateTerraformKube2IAMVariablesEnvironment(o.Shoot.Secret)).
+		Destroy()
+}
+
+// GenerateKube2IAMConfig generates the values which are required to render the chart of kube2iam properly.
+// +deprecated
+func GenerateKube2IAMConfig(o *operation.Operation) (map[string]interface{}, error) {
+	var (
+		enabled = o.Shoot.Kube2IAMEnabled()
+		values  map[string]interface{}
+	)
+
+	if enabled {
+		awsClient := createAWSClient(o.Shoot.Secret, o.Shoot.Info.Spec.Cloud.Region)
+
+		awsAccountID, err := awsClient.GetAccountID()
+		if err != nil {
+			return nil, err
+		}
+
+		v, err := o.InjectShootShootImages(map[string]interface{}{
+			"extraArgs": map[string]interface{}{
+				"base-role-arn": fmt.Sprintf("arn:aws:iam::%s:role/", awsAccountID),
+			},
+		}, "kube2iam")
```

Comment:
Whyn not use the constant you already defined?

Reply:
I removed the constant and most occurrences of `kube2iam` from files other than this one. As it's deprecated anyways and will be removed soon I just used the string.

Reply:
Makes sense, I am just saying because I saw you still have the purpose constant there so I was wondering why no re-name / re-use it.